api/controllers/game/spectate.js

```diff
@@ -1,17 +1,44 @@
 module.exports = async function (req, res) {
   try {
     const { gameId } = req.body;
-    const game = await gameService.populateGame({ gameId });
+    const promiseGame = await gameService.populateGame({ gameId });
+    const promisePlayer = await userService.findUser({ userId: req.session.usr });
 
-    if (game.status || game.players.length < 2) {
-      return res.badRequest({message: 'You can only spectate an ongoing game with two players'});
-    }
-
-    // Subscribe socket to game
-    Game.subscribe(req, [ gameId ]);
-    req.session.spectating = gameId;
-
-    return res.ok(game);
+    Promise.all([promiseGame, promisePlayer])
+      .then(function checkAndSubscribe(values) {
+        const [game, player] = values;
+        if (game.status || game.players.length < 2) {
+          return res.badRequest({ message: 'You can only spectate an ongoing game with two players' });
+        }
+        // Subscribe socket to game
+        const updatePromises = [Game.subscribe(req, [gameId]), (req.session.spectating = gameId)];
+        return Promise.all([game, player, ...updatePromises]);
```

Comment:
I believe both of these are actually synchronous operations so we don't need the `Promise` wrapping here
```suggestion
        Game.subscribe(req, [gameId]);
        req.session.spectating = gameId
```